cumulus/polkadot-parachain/src/chain_spec/glutton.rs

```diff
@@ -15,81 +15,57 @@
 // limitations under the License.
 
 use cumulus_primitives_core::ParaId;
-use parachains_common::AuraId;
 use polkadot_omni_node_lib::chain_spec::{Extensions, GenericChainSpec};
 use sc_service::ChainType;
 use sp_keyring::Sr25519Keyring;
 
-fn glutton_genesis(parachain_id: ParaId, collators: Vec<AuraId>) -> serde_json::Value {
-	serde_json::json!( {
-		"parachainInfo": {
-			"parachainId": parachain_id
-		},
-		"sudo": {
-			"key": Some(Sr25519Keyring::Alice.to_account_id()),
-		},
-		"aura": { "authorities": collators },
-	})
-}
+/// Generic Glutton Westend Config for all currently used setups.
+pub fn glutton_westend_config(
+	para_id: ParaId,
+	chain_type: ChainType,
+	relay_chain: &str,
+) -> GenericChainSpec {
+	let mut properties = sc_chain_spec::Properties::new();
+	properties.insert("ss58Format".into(), 42.into());
 
-pub fn glutton_westend_development_config(para_id: ParaId) -> GenericChainSpec {
-	GenericChainSpec::builder(
-		glutton_westend_runtime::WASM_BINARY.expect("WASM binary was not built, please build it!"),
-		Extensions { relay_chain: "westend-dev".into(), para_id: para_id.into() },
-	)
-	.with_name("Glutton Development")
-	.with_id("glutton_westend_dev")
-	.with_chain_type(ChainType::Local)
-	.with_genesis_config_patch(glutton_genesis(
-		para_id,
-		vec![Sr25519Keyring::Alice.public().into()],
-	))
-	.build()
-}
+	let authorities = match chain_type {
+		ChainType::Development => vec![Sr25519Keyring::Alice.public().into()],
+		// Even the westend live setup does currently use Alice & Bob.
+		_ => vec![Sr25519Keyring::Alice.public().into(), Sr25519Keyring::Bob.public().into()],
+	};
 
-pub fn glutton_westend_local_config(para_id: ParaId) -> GenericChainSpec {
 	GenericChainSpec::builder(
 		glutton_westend_runtime::WASM_BINARY.expect("WASM binary was not built, please build it!"),
-		Extensions { relay_chain: "westend-local".into(), para_id: para_id.into() },
+		Extensions { relay_chain: relay_chain.into(), para_id: para_id.into() },
+	)
+	.with_name(&chain_type_name(para_id, &chain_type))
+	.with_id(&chain_id(para_id, &chain_type))
+	.with_chain_type(chain_type)
+	.with_genesis_config_patch(
+		glutton_westend_runtime::genesis_config_presets::glutton_westend_genesis(
+			authorities,
+			para_id,
+		),
 	)
```

Comment:
Why are you not using the preset here? Why are you calling the code from the runtime here?

Reply:
Because the current implementation wants to pass the para-id from the CLI, but the presets have a hardcoded para-id (see above discussion). If this is no longer the desired behaviour I am happy to change it.

Only glutton has this behaviour, and I remember that some wanted to launch many gluttons on one relay chain for stress testing. So I thought it should remain like that.

Reply:
I just saw the comments above. Maybe we need some `with_genesis_config_patch_and_preset`.

CC @michalkucharczyk

Reply:
Yeah, ideally if presets/patches could be stacked.

But also [`GenesisConfigBuilderRuntimeCaller::get_named_preset`](https://github.com/paritytech/polkadot-sdk/blob/51170495d0a8902bc20e650b17639773698cb0b1/substrate/client/chain-spec/src/genesis_config_builder.rs#L98) could be called to get preset from wasm, and then simply merge the patch:
```
crate::json_patch::merge(&mut config, json!({... para_id}));
```
And use this in `ChainSpecBuilder::with_genesis_config_patch`.

That flow would avoid call into the native runtime.